### PR TITLE
Add ability to publicise group membership

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

**Comment by erikjohnston** [comment]
@matrixbot retest this please

**Comment by erikjohnston** [comment]
@matrixbot retest this please

File: synapse/storage/schema/delta/43/group_server.sql [hunk 1]
```diff
@@ -150,6 +150,7 @@ CREATE TABLE local_group_membership (
     user_id TEXT NOT NULL,
     is_admin BOOLEAN NOT NULL,
     membership TEXT NOT NULL,
+    is_publicised TEXT NOT NULL,  -- if the user is publicising their membership
```
Comment (NegativeMjark) [inline synapse/storage/schema/delta/43/group_server.sql]: Why is this a TEXT column?

**Comment by NegativeMjark** [comment]
Other than the schema oddities it looks good to me?

**Comment by erikjohnston** [comment]
@matrixbot retest this please

**Comment by erikjohnston** [comment]
@matrixbot retest this please

**Comment by erikjohnston** [comment]
@matrixbot retest this please

**Comment by erikjohnston** [comment]
@matrixbot retest this please

**Pull request merged**